File: benchmarks/tsb/bench_series_nlargest.ts
/**
 * Benchmark: Series.nlargest — top-100 elements from a 100k-element Series
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Float64Array.from({ length: ROWS }, (_, i) => Math.sin(i * 0.007) * 500 + i * 0.001);
const s = new Series(data);

for (let i = 0; i < WARMUP; i++) {
  s.nlargest(100);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.nlargest(100);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "series_nlargest",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_series_nlargest.py
"""Benchmark: Series.nlargest — top-100 elements from a 100k-element Series"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = np.sin(np.arange(ROWS) * 0.007) * 500 + np.arange(ROWS) * 0.001
s = pd.Series(data)

for _ in range(WARMUP):
    s.nlargest(100)

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.nlargest(100)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "series_nlargest",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
